turbojpeg.py

```diff
@@ -227,6 +224,69 @@ def encode(self, img_array, quality=85, pixel_format=TJPF_BGR, jpeg_subsample=TJ
         finally:
             self.__destroy(handle)
 
+    """
+    @jpeg_buf           buffer containing jpeg image bytes
+    @scaling_factor     optional image scaling
+    @quality            optional image quality to be reencoded with
+    @flags
+    """
+    def scale_with_quality(self, jpeg_buf, scaling_factor=None, quality=85, flags=0):
+        """decompresstoYUV with scale factor, recompresstoYUV with quality factor"""
+        handle = self.__init_decompress()
+        try:
+            if scaling_factor is not None and \
+                scaling_factor not in self.__scaling_factors:
+                raise ValueError('supported scaling factors are ' +
+                    str(self.__scaling_factors))
+            jpeg_array = np.frombuffer(jpeg_buf, dtype=np.uint8)
+            src_addr = self.__getaddr(jpeg_array)
+            scaled_width, scaled_height, jpeg_subsample, _ = self.__get_header_and_dimensions(
+                handle, jpeg_array, src_addr, scaling_factor)
+            buffer_YUV_size = self.__buffer_size_YUV2(
+                scaled_height, 4, scaled_width, jpeg_subsample)
+            img_array = np.empty([buffer_YUV_size])
+            dest_addr = self.__getaddr(img_array)
+            status = self.__decompressToYUV2(
+                handle, src_addr, jpeg_array.size, dest_addr, scaled_width, 4, scaled_height, flags)
+            if status != 0:
+                self.__report_error(handle)
+                return
+            self.__destroy(handle)
+
+            handle = self.__init_compress()
+            jpeg_buf = c_void_p()
+            jpeg_size = c_ulong()
+            status = self.__compressFromYUV(
+                handle, dest_addr, scaled_width, 4, scaled_height, jpeg_subsample, byref(jpeg_buf),
+                byref(jpeg_size), quality, flags)
+            if status != 0:
+                self.__report_error(handle)
+            dest_buf = create_string_buffer(jpeg_size.value)
+            memmove(dest_buf, jpeg_buf.value, jpeg_size.value)
+            self.__free(jpeg_buf)
+            return dest_buf.raw
+        finally:
+            self.__destroy(handle)
+
+    def __get_header_and_dimensions(self, handle, jpeg_array, src_addr, scaling_factor):
```

Comment:
you don't need jpeg_array here.  Just the size.  Maybe `buffer_size`.  Changing this will make the reusing buffers changes that I mentioned easier.